### PR TITLE
Add tgk11tomsk form example with new block radiogroupwithcontainers

### PR DESCRIPTION
**Pull request by IgorZiegler**
Config example:
```
      "Вариант оплаты": {
        "block": "radiogroupwithdependingdisable",
        "labelWidth": 3,
        "options": [
          {
            "mark": "К оплате за отопление (1/12):<br><small>(с учётом общей задолженности)</small><br>",
            "value": "1/12",
            "checked": true
          },
          {
            "mark": "К оплате за отопление (Факт):<br><small>(с учётом общей задолженности)</small><br>",
            "value": "Факт"
          }
        ],
        "dependingDisable": {
          "1/12": [
            {
              "name": "1/12",
              "block": "text",
              "textAlign": "right",
              "inputWidth": "138px",
              "rightMark": "<span class=\"glyphicon glyphicon-ruble\" aria-hidden=\"true\" style=\"font-size: 11px\"><\/span>",
              "placeholder": 0
            }
          ],
          "Факт": [
            {
              "name": "Факт",
              "block": "text",
              "textAlign": "right",
              "inputWidth": "138px",
              "rightMark": "<span class=\"glyphicon glyphicon-ruble\" aria-hidden=\"true\" style=\"font-size: 11px\"><\/span>",
              "placeholder": 0
            }
          ]
        }
      }
```

### CODE REVIEW

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
+@import "../../styles/_variables.css";
+
+.$(cls-prefix)-form-group {
+
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css]: remove this empty line

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        position: absolute!important;
+        top: 0;
+        left: 0;
+      }
+
+      .radio-field {
+        display: inline-block;
+        margin-top: 14px;
+      }
+      .form-control[disabled] {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css]: add new line before this line

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      margin-bottom: 0;
+      font-weight: normal;
+      cursor: pointer;
+      padding-left: 30px;
+      position: relative;
+      display: inline-block;
+      max-width: 100%;
+
+      .jq-radio {
+        position: absolute!important;
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css]: Shouldn't we try to avoid `!important` if it's possible?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.jade [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    each option in options
+        .radio
+            label
+                input(
+                    type="radio"
+                    name=name
+                    checked=(option.checked && "checked")
+                    value=option.value
+                )
+                !{option.mark}
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.jade]: This line causes following warning: `Warning: missing space before text for line 13 of jade file "/home/user/Development/smartforms/frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.jade"`, you should use `!= option.mark` instead

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+  get templateFn() {
+    return template;
+  }
+
+  get sum() {
+    const checkedItem = this.el.find(`input:radio[name ='${this.config.name}']:checked`) || [];
+
+    return this.config.dependingDisable[(checkedItem && checkedItem.val)
+      ? checkedItem.val()
+      : null].reduce((acc, block) => {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: In the case of
```js
checkedItem && checkedItem.val == false
```
you'll try to get
```js
this.config.dependingDisable[null]
```
Does this makes sense?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css [hunk 6]
```diff
@@ -0,0 +1,27 @@
+@import "../../styles/_variables.css";
+
+.$(cls-prefix)-form-group {
+  .radio {
+    label {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css]: we have styles for same selectors (`.$(cls-prefix)-form-group .radio label`, etc.) in the `radiogroup.css`, shouldn't we make a difference between this elements? Doesn't this styles affects our previous work?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+  onChange() {
+    this.trigger('change', [this, this.value]);
+    this.disableDependencies();
+
+    const checkedItem = this.el.find(`input:radio[name ='${this.config.name}']:checked`) || [];
+
+    this.config.dependingDisable[(checkedItem && checkedItem.val)
+      ? checkedItem.val()
+      : null].forEach((block, index) => {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: https://github.com/hwdtech/smartforms/pull/299#discussion_r72048408

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 8]
```diff
@@ -0,0 +1,96 @@
+import 'jquery.formstyler';
+
+import template from './radiogroupwithdependingdisable.jade';
+
+import './radiogroupwithdependingdisable.css';
+import Base from '../base/base';
+import Factory from '../factory';
+
+export default class RadioGroupWithDependingDisable extends Base {
+
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: remove this empty line

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+    });
+  }
+
+  afterRender() {
+    super.afterRender();
+
+    this.config.options.forEach(option => {
+      this.config.dependingDisable[option.value].forEach(block => {
+        block.afterRender();
+        block.on('change', () => this.trigger('change', [this, this.value]));
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: https://github.com/hwdtech/smartforms/pull/132#discussion_r68389085
https://github.com/hwdtech/smartforms/pull/132#discussion_r68358345

File: public/examples/forms/vp/tgk11tomsk.json [hunk 10]
```diff
@@ -0,0 +1,210 @@
...
+          "Расход2"
+        ]
+      },
+      "К оплате за ГВС",
+      "Пени",
+      "Сумма"
+    ]
+
+
+  }
```
Comment (102) [inline public/examples/forms/vp/tgk11tomsk.json]: remove two empty lines above

File: public/examples/forms/vp/tgk11tomsk.json [hunk 11]
```diff
@@ -0,0 +1,210 @@
...
+        "labelWidth": 3,
+        "value": "п.Зональная станция ул.40 лет Победы д. 3  кв. 44",
+        "disabled": true
+      },
+      "Вариант оплаты": {
+        "block": "radiogroupwithdependingdisable",
+        "labelWidth": 3,
+        "options": [
+          {
+            "mark": "К оплате за отопление (1/12):<br><small>(с учётом общей задолженности)</small><br>",
```
Comment (102) [inline public/examples/forms/vp/tgk11tomsk.json]: Looks like that in the case of basic `radiougpoup` block, analogue of this property called `name`, is this change intended?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+  disableDependencies() {
+    this.config.options.forEach(option => {
+      this.config.dependingDisable[option.value].forEach(block => {
+        block.disable();
+      });
+    });
+  }
+
+  onChange() {
+    this.trigger('change', [this, this.value]);
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: https://github.com/hwdtech/smartforms/pull/299/files#r72052355

File: frontend/blocks/container/container.js [hunk 13]
```diff
@@ -63,6 +63,7 @@ export default class Container extends Base {
   }
 
   set value(val) {
+//    TODO implement this method
```
Comment (102) [inline frontend/blocks/container/container.js]: Shouldn't we follow the same style for all comment blocks? (i.e. this style: https://github.com/hwdtech/smartforms/blob/master/frontend/blocks/form/form.js#L86)

File: frontend/blocks/radiogroupwithdependingdisable/radiogroup_container.js [hunk 14]
```diff
@@ -0,0 +1,37 @@
+import Container from '../container/container';
+import Factory from '../factory';
+import template from './radiogroup_container.jade';
+
+export default class RadioGroupContainer extends Container {
+  afterRender() {
+    super.afterRender();
+
+    this.items.forEach(block => block.on('change', () => this.trigger('change', [this.value])));
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroup_container.js]: Why do we need to pass `this.value` here?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css [hunk 15]
```diff
@@ -0,0 +1,27 @@
+@import "../../styles/_variables.css";
+
+.$(cls-prefix)-form-group {
+  .$(cls-prefix)-radio {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.css]: Why do we want to wrap this class with `.$(cls-prefix)-form-group`?

File: public/examples/forms/vp/tgk11tomsk.json [hunk 16]
```diff
@@ -0,0 +1,208 @@
...
+        ]
+      },
+      "Адрес": {
+        "block": "text",
+        "label": "Адрес",
+        "labelWidth": 3,
+        "value": "п.Зональная станция ул.40 лет Победы д. 3  кв. 44",
+        "disabled": true
+      },
+      "Вариант оплаты": {
```
Comment (102) [inline public/examples/forms/vp/tgk11tomsk.json]: Shouldn't we move this block to the `schema` part of config like other containers?

File: public/examples/forms/vp/tgk11tomsk.json [hunk 17]
```diff
@@ -0,0 +1,208 @@
...
+            "value": "1/12",
+            "checked": true
+          },
+          {
+            "name": "К оплате за отопление (Факт):<br><small>(с учётом общей задолженности)</small><br>",
+            "value": "Факт"
+          }
+        ],
+        "dependingDisable": {
+          "1/12": [
```
Comment (102) [inline public/examples/forms/vp/tgk11tomsk.json]: Why this properties should be arrays?

File: frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+    });
+
+    return this.config.name ? { [this.config.name]: result } : result;
+  }
+
+  set value(val) {
+    // TODO implement this method
+  }
+
+  get enableDependencies() {
```
Comment (102) [inline frontend/blocks/radiogroupwithdependingdisable/radiogroupwithdependingdisable.js]: I'm not sure it's good idea to have verbs in get/set method names

File: public/examples/forms/vp/tgk11tomsk.json (hunk 16 above)
Comment (IgorZiegler) [inline public/examples/forms/vp/tgk11tomsk.json]: It isn't a container

File: frontend/blocks/base/base.js [hunk 19]
```diff
@@ -190,6 +198,10 @@ export default class Base {
     return this.el;
   }
 
+  get disableEl() {
```
Comment (102) [inline frontend/blocks/base/base.js]: `disable` is a verb, shouldn't be used in `get` method

File: frontend/blocks/form/form.js [hunk 20]
```diff
@@ -43,7 +43,8 @@ export default class Form extends Container {
     const summableContainers = [
       'array',
       'fieldset',
-      'table'
+      'table',
+      'radiogroupwithdependingdisable'
```
Comment (102) [inline frontend/blocks/form/form.js]: rename block here too (and PR title should be changed)

File: public/examples/forms/vp/tgk11tomsk.json [hunk 21]
```diff
@@ -0,0 +1,227 @@
...
+      },
+      "Адрес": {
+        "block": "text",
+        "label": "Адрес",
+        "labelWidth": 3,
+        "value": "п.Зональная станция ул.40 лет Победы д. 3  кв. 44",
+        "disabled": true
+      },
+      "Вариант оплаты": {
+        "block": "radiogroupwithdependingdisable",
```
Comment (102) [inline public/examples/forms/vp/tgk11tomsk.json]: old block name too

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 22]
```diff
@@ -0,0 +1,107 @@
...
+
+      return acc + (parseFloat(value[block.name]) || 0);
+    }, 0);
+  }
+
+  get value() {
+    const checkedItem = this.el.find(`input:radio[name ='${this.config.name}']:checked`) || [];
+    const result = {};
+
+    this.activeDependencies.items.forEach(block => Object.assign(result, block.value));
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: use `reduce` here

**Comment by 102** [comment]
@IgorZiegler you should also add basic example of the new field (i.e. without modal window), because we should probably remove vp examples from this repo

File: frontend/blocks/form/form.html [hunk 23]
```diff
@@ -0,0 +1 @@
+<form class="undefined-form undefined"><div class="form-header"><h2></h2></div><div class="clearfix form-content"></div></form>
```
Comment (102) [inline frontend/blocks/form/form.html]: No new line at the end of file (and why do we even want to have this file?)

File: frontend/blocks/text/text.js [hunk 24]
```diff
@@ -17,7 +17,7 @@ export default class Text extends Input {
 
   get value() {
     return {
-      [this.config.name]: this._input.val()
+      [this.config.name]: !this.config.returnNumber ? this._input.val() : parseFloat(this._input.val())
```
Comment (102) [inline frontend/blocks/text/text.js]: `parseFloat(this._input.val())` will return `NaN` if `this._input.val()` isn't a valid string representation of float

**Comment by 102** [comment]
@IgorZiegler https://github.com/hwdtech/smartforms/pull/299#discussion_r72229110 is still actual

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.css [hunk 25]
```diff
@@ -0,0 +1,25 @@
+@import "../../styles/_variables.css";
+
+.radio & .$(cls-prefix)-radio {
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.css]: Does this `&` required here?

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 26]
```diff
@@ -0,0 +1,106 @@
...
+    this.activeDependencies.items.forEach((block, index) => {
+      block.enable();
+      if (0 === index) {
+        block.focus();
+      }
+    });
+  }
+
+  onChange() {
+    this.trigger('change', [this.value]);
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: why do we need `[this.value]`?

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 27]
```diff
@@ -0,0 +1,106 @@
...
+      this.el.find(`#${container.id}`).append(container.el);
+    });
+  }
+
+  afterRender() {
+    super.afterRender();
+
+    this.config.options.forEach(option => {
+      this.dependingDisable[option.value].afterRender();
+      this.dependingDisable[option.value].on('change', () => this.trigger('change', [this.value]));
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: why do we need `[this.value]`?

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 28]
```diff
@@ -0,0 +1,106 @@
...
+  getTemplateData() {
+    const result = super.getTemplateData();
+
+    result.containerOptions = this.containerOptions;
+
+    return result;
+  }
+
+  get sum() {
+    return this.activeDependencies.items.reduce((acc, block) => {
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: `activeDependencies` should be renamed to `activeContainer`

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 29]
```diff
@@ -0,0 +1,106 @@
...
+    this.enableActiveDependencies();
+  }
+
+  disableDependencies() {
+    this.config.options.forEach(option => {
+      this.containerOptions[option.value].disable();
+    });
+  }
+
+  enableActiveDependencies() {
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: `enableActiveContainer` ?

File: frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js [hunk 30]
```diff
@@ -0,0 +1,106 @@
...
+
+    this.config.options.forEach(option => {
+      this.containerOptions[option.value].afterRender();
+      this.containerOptions[option.value].on('change', () => this.trigger('change'));
+    });
+    this.disableDependencies();
+    this.enableActiveDependencies();
+  }
+
+  disableDependencies() {
```
Comment (102) [inline frontend/blocks/radiogroup_with_containers/radiogroup_with_containers.js]: `disableContainers` ?
